### ISSUE TITLE
No build is taking place after adding snapcraft file

### ISSUE THREAD

**Issue by vadi2** [canonical-websites/build.snapcraft.io#823]
I've created https://github.com/vadi2/Mudlet/blob/master/snap/snapcraft.yaml but nothing is happening:

![selection_133](https://cloud.githubusercontent.com/assets/110988/26719897/86b744f8-4786-11e7-8fc9-3ee09c8de40b.png)

**Comment by bartaz** [comment]
Hi, thanks for the report.

I had a quick look at your `snapcraft.yaml` file and it seems it's not valid (I used online http://www.yamllint.com/ validator).

Unfortunately we currently don't detect invalid snapcraft.yaml files which may lead to some confusion like in your case. Sorry for that.
We have it reported (#668) and we will work on giving better feedback on invalid `snapcraft.yaml` files.

**Comment by vadi2** [comment]
That was pretty frustrating - I'm a newbie with yaml and nothing was actually telling me that my intending was just off by two spaces. Instead I was getting very ambiguous errors in any validator I tried - "mapping values are not allowed in this context at line" what does that even mean?

Anyhow, I pass yamllint.com now, waiting on something to happen!

**Comment by bartaz** [comment]
Yes, YAML is whitespace sensitive format and validators' messages are sometimes not very helpful.

After you updated your `snapcraft.yaml` does you repo show as configured now?

**Comment by vadi2** [comment]
It does not - still says "not configured".

**Comment by bartaz** [comment]
I forked your repo to debug it myself.

Indeed, the snapcraft.yaml format seems fine but the snapcraft.yaml is not found by our app.

The GH API we use to fetch the file contents returns 404:
https://api.github.com/repos/vadi2/Mudlet/contents/snap/snapcraft.yaml

It seems to be caused by the fact that master is not default branch in that repo.
We currently build only from master, and while your snapcraft.yaml is in master the GH API uses default branch (which is development not master in your case) and that's why our app mistakenly assumes there is no snapcraft.yaml.

It basically means that we have a bug, because we should make sure we are searching for snapcraft.yaml specifically in a branch that we are trying to build (master in this case) not using GH default that may be different.

I'll report a separate issue for that and we will try to fix it soon.

For a temporary workaround I can suggest changing your default branch to master in your fork. This should let build.snapcraft.io correctly detect the file.

Thanks again for the report.

**Comment by vadi2** [comment]
Got it, thanks! That now set it to configured.

**Comment by bartaz** [comment]
Sorry to hear that.

If your repo shows as configured now it means that from build.snapcraft.io point of view snapcraft.yaml is valid and detected. If you also have a name registered the build should be triggered.
If there is anything else wrong in snapcraft.yaml build would fail, but there would be a build and build logs.

So, do you see you repo both as configured (a tick in Configured column) and registered? Does your build status is 'Never built' or something else?

If it's never built you could try pushing a new commit to trigger the webhook.
You could also try to register the name again (even exactly the same name) - that should re-trigger build request.

If that doesn't help we may need to investigate more.

**Comment by vadi2** [comment]
It finally built it 13 hours ago - so there was a really long delay. Thanks for the help and your work on the site!
